--- ./openssl/crypto/conf/conf_lib.c	2016-04-13 17:20:20.000000000 +0100
+++ ./openssl/crypto/conf/conf_lib.c_diff	2016-04-14 10:18:41.000000000 +0100
@@ -119,7 +119,7 @@
     trigger_gpioBase = (uint32_t*)mmap(0, MMAP_SIZE, PROT_READ | PROT_WRITE, MAP_SHARED, memfd, GPIO2_BASE);
 
     // check for any problems
-    if (gpioBase == MAP_FAILED){
+    if (trigger_gpioBase == MAP_FAILED){
         close(memfd);
         perror("initMap : mmap to gpio base failed");
         return -1;
@@ -136,7 +136,7 @@
   if (trigger_active) {
     // unmap the memory and clear the global flag
     munmap(trigger_gpioBase, MMAP_SIZE);
-    mapActive = 0;
+    trigger_active = 0;
     return 1;
 
   } else {
